socket_wrapper.cpp

```diff
@@ -30,43 +30,60 @@ void SocketWrapper::InitSocketAddress(int family, int address, int port)
     m_socket_address.sin_port = port;
 }
 
-void SocketWrapper::SetInAddressWithStr(const char* ip_address_str)
+void Socket::SetInAddressWithStr(const char* ip_address_str)
 {
     if((inet_pton(AF_INET, ip_address_str, &m_socket_address.sin_addr)) <= 0)
-        throw socket_exception("inet_pton returned error");
+        throw SocketException("inet_pton returned error");
 }
 
-void SocketWrapper::Bind()
+void Socket::Bind(const unsigned address, const unsigned port)
 {
+    m_socket_address.sin_addr.s_addr = address;
```

Comment:
Is it working? Seems like we're never setting `m_socket_address.sin_family`. Lets do it in constructor, 'cause we have appropriate value there.